signer/src/testing/block_observer.rs

```diff
@@ -60,13 +61,29 @@ impl TestHarness {
     }
 
     /// Get the deposit transactions in the test harness.
-    pub fn deposits(&self) -> &HashMap<Txid, GetTxResponse> {
+    pub fn deposits(&self) -> &HashMap<Txid, (GetTxResponse, BitcoinTxInfo)> {
         &self.deposits
     }
 
     /// Add a single deposit transaction to the test harness.
     pub fn add_deposit(&mut self, txid: Txid, response: GetTxResponse) {
-        self.deposits.insert(txid, response);
+        let tx_info = BitcoinTxInfo {
+            in_active_chain: response.block_hash.is_some(),
+            fee: bitcoin::Amount::from_sat(1000),
+            tx: response.tx.clone(),
+            txid: response.tx.compute_txid(),
+            hash: response.tx.compute_wtxid(),
+            size: response.tx.vsize() as u64,
```

Comment:
nit: (maybe?)
```suggestion
            size: response.tx.total_size() as u64,
```

Reply:
Done in https://github.com/stacks-network/sbtc/pull/675/commits/0902b2a9b7a79745473e7b90405ef5e4860113d7.